### PR TITLE
update cli

### PR DESCRIPTION
**Pull request by auanmu**
added some functions for the commands and put the code inside them

### CODE REVIEW

File: MVC/view/CLI.py [hunk 1]
```diff
@@ -76,8 +76,80 @@ def showHoneyComb(self):
...
+                self.controller.controllerSaveGame(inputFile)
+            else:
+                print("Ok, lets generate a new puzzle! ")
+        self.controller.controllerNewGame()
+        isAuto = input("Do you want it to be automatically generated? (yes/no): ")
+        while isAuto.lower() != "yes" and isAuto.lower() != "no":
+            isAuto = input("Do you want it to be automatically generated? (yes/no): ")
+    
+        if (isAuto.lower() == "yes"):
+            self.controller.controllerRunAutoGame()
```
Comment (tmhughes1) [inline MVC/view/CLI.py:94]: These 5 lines could be refactored since they appear again below (just use a parameter to check for AutoGame or BaseGame

File: MVC/view/CLI.py [hunk 2]
```diff
@@ -76,8 +76,80 @@ def showHoneyComb(self):
...
+                    self.check = self.controller.controllerCheckPangram(userInput)
+            self.controller.controllerRunBaseGame(userInput)
+            print("User letters: " + self.controller.controllerGetLetters())
+            print("Required letter: " + self.controller.controllerGetReqLetter())
+            self.showHoneyComb()
+            self.controller.controllerUpdatePuzzleState1()
 
-         
+    def showPuzzle(self):
+        if (self.controller.controllerGetPuzzleState() == 0):
```
Comment (tmhughes1) [inline MVC/view/CLI.py:112]: These two lines can be refactored since they're used in quite a few of the commands!

**Review by tmhughes1 — CHANGES_REQUESTED** [review]
Check my comments: there's some code that can easily be placed into a separate function (refactoring!) since it's used more than once! If you need help with this let me know!

File: MVC/view/CLI.py [hunk 3]
```diff
@@ -76,8 +94,68 @@ def showHoneyComb(self):
...
+                self.controller.controllerSaveGame(inputFile)
+        inputFile = input("Enter the name of the file you want to load: ")
+        checkFile = inputFile + ".json"
+        if os.path.exists(checkFile):
+            self.controller.controllerGameLoadCLI(inputFile)
+            print("Puzzle loaded!")
+            self.showHoneyComb()
+        else:
+            print("Uh-oh! Couldn't find that file. Reenter the load command and try again.")
+    def checkGameStarted(self):
```
Comment (tmhughes1) [inline MVC/view/CLI.py:156]: Why did you add this ?

Comment (tmhughes1) [reply MVC/view/CLI.py:156]: referring to checkGameStarted

File: MVC/view/CLI.py [hunk 4]
```diff
@@ -76,8 +94,68 @@ def showHoneyComb(self):
...
+
+    def showPuzzle(self):
+        if (self.controller.controllerGetPuzzleState() == 0):
+            print("No game started!")
+        else:
+            print("Your letters: " + self.controller.controllerGetLetters())
+            print("Required letter: " + self.controller.controllerGetReqLetter())
+            print("Guessed words: " + str(self.controller.controllerGetGuessedWordsCLI()))
+            self.showHoneyComb()
+    def showFoundWords(self):
```
Comment (tmhughes1) [inline MVC/view/CLI.py:122]: Please add consisten spacing.

File: MVC/view/CLI.py [hunk 5]
```diff
@@ -76,8 +94,68 @@ def showHoneyComb(self):
...
+            print("Guessed words: " + str(self.controller.controllerGetGuessedWordsCLI()))
+    
+    def shuffleLetters(self):
+        if (self.controller.controllerGetPuzzleState() == 0):
+                print("No game started!")
+        else:
+            self.controller.controllerShuffleAuto()
+            self.showHoneyComb()
+            print("Your letters: " + self.controller.controllerGetLetters())
+    def savePuzzle(self):
```
Comment (tmhughes1) [inline MVC/view/CLI.py:135]: Spacing again here.. Also we need to be using consistent commenting. Using the ''' what the method does blah blah''' format.

File: MVC/view/CLI.py [hunk 6]
```diff
@@ -124,81 +202,19 @@ def startGame(self):
...
-                            print("Your letters: " + self.controller.controllerGetLetters())
+                    self.shuffleLetters()
                 case "savepuzzle":
-                        if (self.controller.controllerGetPuzzleState() == 0):
-                            print("No game started!")
-                        else:
-                            inputFile = input("Please enter a name for the file: ")
-                            self.controller.controllerSaveGame(inputFile)
+                    self.savePuzzle()
                 case "loadpuzzle":
```
Comment (tmhughes1) [inline MVC/view/CLI.py:218]: uhh didn't you make a function for this above like the rest of the commands? why is the code still here...

**Review by tmhughes1 — CHANGES_REQUESTED** [review]
Please see my latest comments!

File: MVC/view/CLI.py [hunk 7]
```diff
@@ -104,13 +114,15 @@ def newPuzzle(self):
             else:
                 print("Ok, lets generate a new puzzle! ")
         self.controller.controllerNewGame()
-        isAuto = input("Do you want it to be automatically generated? (yes/no): ")
+        isAuto   = input("Do you want it to be automatically generated? (yes/no): ")
```
Comment (tmhughes1) [inline MVC/view/CLI.py:117]: why'd you add an extra space here? please remove!

**Review by tmhughes1 — CHANGES_REQUESTED** [review]
New functions (and their comments) should have a newline before them instead of just being put right after, for readability ! Please fix.

**Review by tmhughes1 — APPROVED** [review]
OK I think this is good now, thank you for entertaining my requested changes!

**Review by SteamsDev — APPROVED** [review]
Tested this before approving and did not run into any issues.
